components/datatypes/mobile-application.json

```diff
@@ -0,0 +1,133 @@
+{
+  "meta:license": [
+    "Copyright 2020 Adobe. All rights reserved.",
+    "This work is licensed under a Creative Commons Attribution 4.0 International (CC BY 4.0) license",
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/xdm/context/mobile-application",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "Mobile Application",
+  "type": "object",
+  "description": "Details related to mobile application generated interactions. Mobile Application refers to a mobile software experience that can be installed, run, closed, or uninstalled by an end user. Details additionally include session and application lifecycle metrics reported by the application at the time of the experience.",
+  "meta:extensible": true,
+  "definitions": {
+    "mobileApplication": {
+      "properties": {
+        "xdm:id": {
+          "title": "Application identifier",
+          "type": "string",
+          "description": "Identifier of the application."
+        },
+        "xdm:name": {
+          "title": "Name",
+          "type": "string",
+          "description": "Name of the application."
+        },
+        "xdm:version": {
+          "title": "Version",
+          "type": "string",
+          "description": "Version of the application."
+        },
+        "xdm:isLaunch": {
+          "title": "Is launch event",
+          "type": "boolean",
+          "description": "Indicates whether or not this is an application launch event."
+        },
+        "xdm:launchType": {
+          "type": "string",
+          "description": "Specifies the launch event type. Install indicates this is the first application launch after the application was installed. Upgrade indicates this is the first application launch after the application was upgraded. Launch is an application launch which is not after an application install or upgrade.",
+          "enum": ["launch", "install", "upgrade"],
+          "meta:enum": {
+            "launch": "Launch",
+            "install": "Install",
+            "upgrade": "Upgrade"
+          }
+        },
+        "xdm:isClose": {
```

Comment:
Would you consider define the events as "measure" datatype as that is the preferred way of defining events today on the platform. That way, analytics will have a consistent manner to process all events on the platform.